src/bosh_azure_cpi/lib/cloud/azure/vm_manager.rb

```diff
@@ -183,16 +190,23 @@ def get_load_balancer(resource_pool)
     def create_network_interfaces(instance_id, location, resource_pool, network_configurator)
       network_interfaces = []
       load_balancer = get_load_balancer(resource_pool)
-      public_ip = get_public_ip(network_configurator.vip_network)
       networks = network_configurator.networks
       networks.each_with_index do |network, index|
-        security_group = get_network_security_group(resource_pool, network)
         nic_name = "#{instance_id}-#{index}"
+        public_ip = nil
+        if network.vip?
+          public_ip = get_public_ip(network_configurator.vip_network)
+        elsif resource_pool['assign_dynamic_public_ips']
+          @logger.info "Assigning public IP for NIC: #{nic_name}"
+          public_ip = create_public_ip(nic_name, location)
+        end
+        security_group = get_network_security_group(resource_pool, network)
+
         nic_params = {
           :name                => nic_name,
           :location            => location,
           :private_ip          => (network.is_a? ManualNetwork) ? network.private_ip : nil,
-          :public_ip           => index == 0 ? public_ip : nil,
+          :public_ip           => public_ip,
```

Comment:
possibly this code will assign the same public ip to different nics when there are multiple nics.